backend/fms_core/tests/test_services/sample_next_step_test.py

```diff
@@ -497,3 +500,115 @@ def test_execute_workflow_action(self):
+                
+        letter_B = "B"
+        start = 1
+        end = 7
+        study_B, _ = Study.objects.get_or_create(letter=letter_B,
+                                       project=self.project,
+                                       workflow=self.workflow_pcr_free,
+                                       start=start,
+                                       end=end)
+
+        protocol_1 = Protocol.objects.get(name="Extraction")
+        process_1 = Process.objects.create(protocol=protocol_1)
+        process_measurement_1 = ProcessMeasurement.objects.create(process=process_1,
+                                                                  source_sample=sample_in,
+                                                                  execution_date=datetime.date(2021, 1, 10),
+                                                                  volume_used=5000)
+        SampleLineage.objects.create(parent=sample_in, child=sample_out, process_measurement=process_measurement_1)
+
+        old_sample_to_study_workflow_1, _, _ = queue_sample_to_study_workflow(sample_in, study_B)
+
+        if order == 0:
+             return study_B, step_1, sample_in, process_measurement_1, sample_out
+
+        errors, warnings = execute_workflow_action(workflow_action=WorkflowAction.NEXT_STEP.label,
+                                                   step=step_1,
+                                                   current_sample=sample_in,
+                                                   process_measurement=process_measurement_1,
+                                                   next_sample=sample_out)
+        self.assertEquals(errors, [])
+        self.assertEquals(warnings, [])
+        
+        if order == 1:
+             return study_B, step_1, sample_in, process_measurement_1, sample_out
+
+        step_2 = Step.objects.get(name="Sample QC")
+        protocol_2 = Protocol.objects.get(name="Sample Quality Control")
+        process_2 = Process.objects.create(protocol=protocol_2)
+        process_measurement_2 = ProcessMeasurement.objects.create(process=process_2,
+                                                                  source_sample=sample_out,
+                                                                  execution_date=datetime.date(2021, 1, 10),
+                                                                  volume_used=2) 
+
+        errors, warnings = execute_workflow_action(workflow_action=WorkflowAction.NEXT_STEP.label,
+                                            step=step_2,
+                                            current_sample=sample_out,
+                                            process_measurement=process_measurement_2)
+        self.assertEquals(errors, [])
+        self.assertEquals(warnings, [])
+
+        return study_B, step_2, sample_out, process_measurement_2, sample_out
+
+    def test_dequeue_sample_from_specific_step_and_update_step_history(self):
+        for order in [1, 2, 3]:
```

Comment:
... in range(3) ?

Reply:
It's mainly semantics. There's no such thing as order 0.